bot/skills/covid_mode.py

```diff
@@ -0,0 +1,364 @@
+QUARANTIN_MUTE_DURATION = timedelta(hours=12)
+
+DAILY_INFECTION_RATE = 0.01
+
+COUGH_INFECTION_CHANCE_MASKED = 0.01
+COUGH_INFECTION_CHANCE_UNMASKED = 0.3
+
+RANDOM_COUGH_INFECTED_CHANCE = 0.1
+RANDOM_COUGH_UNINFECTED_CHANCE = 0.002
+
+INFECTION_CHANCE_MASKED = 0.01
+INFECTION_CHANCE_UNMASKED = 0.15
+
+LETHALITY_RATE = 0.8
+
+JOB_QUEUE_DAILY_INFECTION_KEY = 'covid_daily_infection_job'
+JOB_QUEUE_REPEATING_COUGHING_KEY = 'covid_repeating_coughing_job'
+
+REPEATING_COUGHING_INTERVAL = timedelta(minutes=1)
+
+DAILY_INFECTION_TIME = time(hour=0, minute=0, second=0, microsecond=0, tzinfo=None)
+
+class DB:
+    """
+        Members document:
+        {
+            _id: 420,                                   # int       -- tg user id
+            meta: {...},                                # Dict      -- full tg user object (just in case)
+            infected_since: datetime(...),              # DateTime  -- time of infection
+            quarantined_since: datetime(...),           # DateTime  -- time of infection
+            quarantined_until: datetime(...),           # DateTime  -- time of infection
+        }
+
+        Settings document:
+        {
+            covidstatus: True
+        }
+    """
+
+    def __init__(self, db_name: str):
+        self._coll: Collection = get_db(db_name).members
+        self._setts: Collection = get_db(db_name).settings
+
+    def find_all(self):
+        return list(self._coll.find({}))
+
+    def find(self, user_id: str):
+        return self._coll.find_one({"_id": user_id})
+
+    def add(self, user: User):
+        return self._coll.update_one({
+            "_id": user.id,        
+        }, {
+            "$set": {"meta": user.to_dict()}
+        }, upsert=True)
+
+    def get_covidstatus(self):
+        setts = self._setts.find_one({})
+        if setts is None:
+            return False
+        return setts['covidstatus']
+    
+    def set_covidstatus(self, status):
+        self._setts.update_one({}, {
+            "$set": {"covidstatus": status}
+        })
+
+    def infect(self, user_id: str):
+        self._coll.update_one({"_id": user_id}, {
+            "$set": {"infected_since": datetime.now()}
+        })
+
+    def is_user_infected(self, user_id: str):
+        return self._coll.find_one({
+            "_id": user_id, 
+            "infected_since": { "$exists": True }
+        }) is not None
+
+    def add_quarantine(self, user_id: str, since: datetime, until: datetime):
+        self._coll.update_one({"_id": user_id}, {
+            "$set": {
+                "quarantined_since": since,
+                "quarantined_until": until
+            }
+        })
+
+    def get_quarantined(self):
+        self._coll.find({ "quarantined_until": { "$lte": datetime.now() } })
+
+    def dead(self, user_id: str, mute_min: int):
+        self._coll.update_one({"_id": user_id}, {
+            "$inc": {"shot_counter": 1, "dead_counter": 1, "total_time_in_club": mute_min * 60},
```

Comment:
time_in_club is in seconds?

Reply:
looks like unrelated copy|paste code from another mode 😐 😐 😐